Core/StorageInconsistencyMonitor.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  StorageInconsistencyMonitor.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+public protocol AppActivationInconsistencyMonitoring {
+    func didBecomeActive(isProtectedDataAvailable: Bool)
+}
+
+public protocol StatisticsStoreInconsistencyMonitoring {
+    func statisticsDidLoad(hasFileMarker: Bool, hasInstallStatistics: Bool)
+}
+
+public protocol AdAttributionReporterInconsistencyMonitoring {
+    func addAttributionReporter(hasFileMarker: Bool, hasCompletedFlag: Bool)
+}
+
+public struct StorageInconsistencyMonitor: AppActivationInconsistencyMonitoring & StatisticsStoreInconsistencyMonitoring & AdAttributionReporterInconsistencyMonitoring {
```

Comment:
Can I suggest documenting what type of storage inconsistencies we're trying to capture?  Doesn't need to be long, just an overview of where this is coming from, and why this is added.

I think that'd be of great help to aid future maintainability.

Reply:
https://github.com/duckduckgo/iOS/pull/3510/commits/9f039368d90fc71c46ae130ec75a92e77d858b92